app/views/shared/_header.html.haml

```diff
@@ -1,22 +1,40 @@
 %section#top-header.row
   %h1.col-md-2
     = link_to "MyFLiX"
-  %ul.col-md-4.clearfix
-    %li= link_to "Videos"
-    %li= link_to "My Queue"
-    %li= link_to "People"
-  %form.col-md-5.navbar-form(action="")
-    .form-group
-      %input.form-control(type="text" placeholder="Search for videos here")
-    %button.btn.btn-default(type="submit") Search
-  #user_links.pull-right
-    %ul
-      %li.dropdown
-        %a(href="#" id="dlabel" role="button" data-toggle="dropdown" class="dropdown-toggle" data-target="#")
-          Welcome, Philip J. Fry
-          %b.caret
-        %ul.dropdown-menu(role="menu" aria-labelledby="dlabel")
-          %li
-            %a(href="#") Account
-            %a(href="#") Plan and Billing
-            %a(href="#") Sign Out
+  - if signed_in?
+    %ul.col-md-4.clearfix
+      %li= link_to "Videos"
+      %li= link_to "My Queue"
+      %li= link_to "People"
+    / %form.col-md-5.navbar-form(action="")
+    /   .form-group
+    /     %input.form-control(type="text" placeholder="Search for videos here")
+    /   %button.btn.btn-default(type="submit") Search
+    /   .form-group 
+    = form_tag search_videos_path, method: 'get', class: "span5 for-search" do
+      = text_field_tag "query", nil, placeholder: "Search for videos here", class: "search-query form-control"
+      = submit_tag "Search", type: 'submit', class: "button btn btn-default"
+
+    #user_links.pull-right
+      %ul
+        %li.dropdown
+          %a(href="#" id="dlabel" role="button" data-toggle="dropdown" class="dropdown-toggle" data-target="#")
+            = current_user.full_name
+            %b.caret
+          %ul.dropdown-menu(role="menu" aria-labelledby="dlabel")
+            %li
+              %a(href="#") Account
+              %a(href="#") Plan and Billing
+              = link_to "Sign Out", sign_out_path
+
+/ #user_links.pull-right
```

Comment:
I would recommend not getting into the habit of keeping commented code around. It becomes clutter that never gets removed because nobody knows why it was kept.

Reply:
I think this should be the second time you are pointing this out to me. I sort of developed that bad habit of not wanting to let old code go. Would do better